### PR TITLE
[Merged by Bors] - feat: port Analysis.SpecialFunctions.Arsinh

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Arsinh.lean [hunk 1]
```diff
@@ -19,21 +19,21 @@ inverse, arsinh.
 
 ## Main definitions
 
-- `real.arsinh`: The inverse function of `real.sinh`.
+- `Real.arsinh`: The inverse function of `Real.sinh`.
 
-- `real.sinh_equiv`, `real.sinh_order_iso`, `real.sinh_homeomorph`: `real.sinh` as an `equiv`,
-  `order_iso`, and `homeomorph`, respectively.
+- `Real.sinhEquiv`, `Real.sinhOrderIso`, `Real.sinhHomeomorph`: `Real.sinh` as an `Equiv`,
+  `OrderIso` and `Homeomorph`, respectively.
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Arsinh.lean:25]: Don't you like Oxford comma?

File: Mathlib/Analysis/SpecialFunctions/Arsinh.lean [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+open Function Filter Set
+
+open scoped Topology
+
+namespace Real
+
+variable {x y : ℝ}
+
+/-- `arsinh` is defined using a logarithm, `arsinh x = log (x + sqrt(1 + x^2))`. -/
+-- @[pp_nodot]
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Arsinh.lean:55]:
```suggestion
-- @[pp_nodot] is no longer needed
```

File: Mathlib/Analysis/SpecialFunctions/Arsinh.lean [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+@[simps]
+def sinhEquiv : ℝ ≃ ℝ where
+  toFun := sinh
+  invFun := arsinh
+  left_inv := arsinh_sinh
+  right_inv := sinh_arsinh
+#align real.sinh_equiv Real.sinhEquiv
+
+/-- `Real.sinh` as an `OrderIso`. -/
+@[simps! (config := { fullyApplied := false })]
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Arsinh.lean:117]:
```suggestion
@[simps! (config := .asFn)]
```

File: Mathlib/Analysis/SpecialFunctions/Arsinh.lean [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+def sinhOrderIso : ℝ ≃o ℝ where
+  toEquiv := sinhEquiv
+  map_rel_iff' := @sinh_le_sinh
+#align real.sinh_order_iso Real.sinhOrderIso
+
+/-- `Real.sinh` as a `Homeomorph`. -/
+@[simps! (config := { fullyApplied := false })]
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Arsinh.lean:124]:
```suggestion
@[simps! (config := .asFn)]
```

File: Mathlib/Analysis/SpecialFunctions/Arsinh.lean [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+def arsinh (x : ℝ) :=
+  log (x + sqrt (1 + x ^ 2))
+#align real.arsinh Real.arsinh
+
+theorem exp_arsinh (x : ℝ) : exp (arsinh x) = x + sqrt (1 + x ^ 2) := by
+  apply exp_log
+  rw [← neg_lt_iff_pos_add']
+  calc
+    -x ≤ sqrt (x ^ 2) := le_sqrt_of_sq_le (by linarith)
+    _ < sqrt (1 + x ^ 2) := sqrt_lt_sqrt (sq_nonneg _) (lt_one_add _)
```
Comment (urkud) [inline Mathlib/Analysis/SpecialFunctions/Arsinh.lean:65]:
```suggestion
  apply lt_sqrt_of_sq_lt
  simp
```

**Review by urkud — COMMENTED** [review]
I have several optional suggestions, otherwise LGTM.
bors d+

**Comment by Parcly-Taxel** [comment]
bors merge

**Pull request closed without merging**